internal/util/xrh_header_parser.go

```diff
@@ -0,0 +1,81 @@
+package util
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/redhatinsights/platform-go-middlewares/identity"
+	"github.com/sirupsen/logrus"
+)
+
+func ParseXRHIdentityHeader(identityHeader string) (*identity.XRHID, error) {
+	var XRHIdentity identity.XRHID
+	decodedIdentity, err := base64.StdEncoding.DecodeString(identityHeader)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding Identity: %v", err)
+	}
+
+	err = json.Unmarshal(decodedIdentity, &XRHIdentity)
+
+	if err != nil {
+		logrus.Errorf("x-rh-identity header is not a valid json: %s. Identity: %s", err.Error(), identityHeader)
+		return nil, fmt.Errorf("x-rh-identity header is not a valid json: %s", err.Error())
+	}
+
+	// XRHIdentity.Identity.User.UserID
+	return &XRHIdentity, nil
+}
+
+type DecodedToken struct {
+	UserId        string `json:"user_id"`
+	OrgId         string `json:"org_id"`
+	AccountNumber string `json:"account_number"`
+	Username      string `json:"username"`
+}
+
+// Parse JWT token without decode key
+func ParseJWTToken(tokenString string) (DecodedToken, error) {
+	str := tokenString
+	str = strings.Split(str, ".")[1]
+	str = strings.ReplaceAll(str, "-/", "+")
+	str = strings.ReplaceAll(str, "_/", "/")
+
+	switch len(str) % 4 {
```

Comment:
We should add some unit tests to verify the different possibilities (that will help document what the function is doing exactly).